lib/ovirt/service.rb

```diff
@@ -127,11 +128,13 @@ def status(link)
     end
 
     def api_uri(path = nil)
-      uri = "#{base_uri}/api"
+      uri = "#{base_uri}"
+      uri += api_version == 'v4' ? '/ovirt-engine/api' : '/api'
       unless path.nil?
         parts = path.to_s.split('/')
-        parts.shift if parts.first == ''    # Remove leading slash
-        parts.shift if parts.first == 'api' # We already have /api in our URI
+        parts.shift if parts.first == ''             # Remove leading slash
+        parts.shift if parts.first == 'ovirt-engine' # We already have /ovirt-engine in our URI
+        parts.shift if parts.first == 'api'          # We already have /api in our URI
         uri += "/#{parts.join('/')}" unless parts.empty?
```

Comment:
Would it be possible to use a URI builder class for this?  I think we can change the base_uri method to return the URI object, and then that object can be manipulated instead of all this String manipulation.